src/coreclr/vm/mlinfo.cpp

```diff
@@ -1163,7 +1163,7 @@ MarshalInfo::MarshalInfo(Module* pModule,
     m_byref = TRUE;
 #endif
 
-
+    const bool useBuiltInMarshalling = ms != MARSHAL_SCENARIO_NDIRECT || pModule->IsRuntimeMarshallingEnabled();
 
     // Retrieve the native type for the current parameter.
```

Comment:
Would it be better to just check that all signature arguments are blittable, and do not even bother parsing the NativeTypeInfo?

Reply:
I've refactored this to be cleaner and short-circuit out when `useBuiltInMarshalling` is `false`.